### PR TITLE
Sanitize and Escape refs in git backend (#21464)

### PR DESCRIPTION
**Pull request by 6543**
Backport #21464 and #21465

### CODE REVIEW

**Review by noerw — APPROVED** [review]

File: modules/git/git.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ import (
 )
 
 // GitVersionRequired is the minimum Git version required
-const GitVersionRequired = "2.0.0"
+const GitVersionRequired = "2.20.0"
```
Comment (6543) [inline modules/git/git.go]:
```suggestion
const GitVersionRequired = "2.0.0"
```

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

File: modules/gitgraph/graph.go [hunk 2]
```diff
@@ -24,35 +24,30 @@ func GetCommitGraph(r *git.Repository, page, maxAllowedColors int, hidePRRefs bo
...
 	}
 
-	args := make([]string, 0, 12+len(branches)+len(files))
-
-	args = append(args, "--graph", "--date-order", "--decorate=full")
+	graphCmd := git.NewCommand(r.Ctx, "log", "--graph", "--date-order", "--decorate=full")
 
 	if hidePRRefs {
-		args = append(args, "--exclude="+git.PullPrefix+"*")
+		graphCmd.AddArguments("--exclude=" + git.PullPrefix + "*")
```
Comment (delvh) [inline modules/gitgraph/graph.go:30]: … We certainly have to agree at some point whether we want spaces around string concatenation, or not.
Some maintainers (like me) prefer spaces, and others don't want them. I've seen such changes now too often…

**Review by delvh — APPROVED** [review]

File: modules/gitgraph/graph.go (hunk 2 above)
Comment (wxiaoguang) [reply modules/gitgraph/graph.go:30]: That's the correct Golang format.



Golang has different space styles for `append()` and function call.


<details>


image

</details>

Comment (wxiaoguang) [reply modules/gitgraph/graph.go:30]: Golang has different styles for `append()` or and function call.

Comment (6543) [reply modules/gitgraph/graph.go:30]: yes ... and in this case as long as the ci dont complain we should leave it as is

**Pull request merged**